requirements_override.nix

```diff
@@ -2,6 +2,14 @@
 
 self: super: {
   lxml = python.overrideDerivation super.lxml (old: {
-    propagatedNativeBuildInputs = [ ];
+    # This is to avoid the next error:
+    #
+    # $ nix-shell --show-trace
+    # error: while evaluating the attribute ‘propagatedBuildInputs’ of the derivation ‘python3.5-heutagogy-1.0’ at <nixpkgs>/pkgs/stdenv/generic/make-derivation.nix:98:11:
+    # while evaluating the attribute ‘propagatedBuildInputs’ of the derivation ‘python3.5-newspaper3k-0.1.9’ at <nixpkgs>/pkgs/stdenv/generic/make-derivation.nix:98:11:
+    # while evaluating the attribute ‘propagatedBuildInputs’ of the derivation ‘python3.5-beautifulsoup4-4.6.0’ at <nixpkgs>/pkgs/stdenv/generic/make-derivation.nix:98:11:
+    # while evaluating the attribute ‘propagatedBuildInputs’ of the derivation ‘python3.5-python3.5-lxml-4.1.1’ at <nixpkgs>/pkgs/stdenv/generic/make-derivation.nix:98:11:
+    # infinite recursion encountered, at undefined position
+    propagatedBuildInputs = [ ];
```

Comment:
What the difference between `propagatedNativeBuildInputs` and `propagatedBuildInputs`?

Reply:
For native builds, there is no difference as they are both merged together.

The difference only manifests when cross-compiling: `nativeBuildInputs` are built for the host and `buildInputs` are built for the target. To put it simply, `native` is for tools that you want to run when you're building the package (compiler, git, patches, configuration, etc.), plain `buildInputs` is for libraries and what's going to be used when an application runs.